openpype/hosts/houdini/plugins/publish/extract_active_view_thumbnail.py

```diff
@@ -0,0 +1,55 @@
+import pyblish.api
+
+from openpype.pipeline import publish, registered_host
+from openpype.hosts.houdini.api import lib
+from openpype.hosts.houdini.api.pipeline import IS_HEADLESS
+
+
+class ExtractActiveViewThumbnail(publish.Extractor):
+    """Set instance thumbnail to a screengrab of current active viewport.
+
+    This makes it so that if an instance does not have a thumbnail set yet that
+    it will get a thumbnail of the currently active view at the time of
+    publishing as a fallback.
+
+    """
+    order = pyblish.api.ExtractorOrder + 0.49
+    label = "Extract Active View Thumbnail"
+    families = ["workfile"]
+    hosts = ["houdini"]
+
+    def process(self, instance):
+        if IS_HEADLESS:
+            self.log.debug(
+                "Skip extraction of active view thumbnail, due to being in"
+                "headless mode."
+            )
+            return
+
+        thumbnail = instance.data.get("thumbnailPath")
+        self.log.debug(thumbnail)
+        if not thumbnail:
+            view_thumbnail = self.get_view_thumbnail(instance)
+            if not view_thumbnail:
+                return
+
+            self.log.debug("Setting instance thumbnail path to: {}".format(
+                view_thumbnail
+            ))
+            instance.data["thumbnailPath"] = view_thumbnail
+
+    def get_view_thumbnail(self, instance):
+
+        host = registered_host()
+        current_filepath = host.get_current_workfile()
```

Comment:
Consider to use `currentFile` from context data? It is still valid as you can find the use of currentFile in workfile [workfile collector](https://github.com/ynput/OpenPype/blob/8501da4c09123ea933b062e3c80b9f22055282b4/openpype/hosts/houdini/plugins/publish/collect_workfile.py#L16C4-L16C4)
```suggestion
        current_filepath = instance.context.data["currentFile"]
```

Reply:
This is staging/temporary publish file - solely for publihing. I'd personally not store this next to the workfile but just in the staging dir. @iLLiCiTiT thoughts?